### ISSUE TITLE
Convert GCN project to pytorch

### ISSUE THREAD

**Issue by zhangyu2ustc** [SIMEXP/fmri_predict#19]
deep learning on graph

pytorch has very nice tools for gcns, including: 
DGL:                           https://github.com/dmlc/dgl
PyTorch Geometric :  https://github.com/rusty1s/pytorch_geometric

I want to transfer my GCN models using these new tools

**Comment by zhangyu2ustc** [comment]
First, I need to rewrite a dataset class which loads fmri data and event data:
check the notebook for examples:    pytorch_hcp_task_fmri.ipynb

'''
#######
params = {'batch_size': 10,
          'shuffle': True,
          'num_workers': 2}

ind = np.loadtxt("motor_perm_indices.txt", comments="#", delimiter=",", unpack=False).astype(int)

fmri_matrix = np.array(subjects_tc_matrix)
label_matrix = np.array(subjects_trial_label_matrix)
fmri_dataset = HCP_taskfmri_matrix_datasets(pathout,fmri_matrix,label_matrix,target_name)
#fmri_dataset = HCP_taskfmri_matrix_datasets(pathout,fmri_files,label_matrix,target_name)
                     
fmri_loader = DataLoader(fmri_dataset, collate_fn=fmri_samples_collate_fn, **params)
data,label = next(iter(fmri_loader))
print(data.size(),label.size())
'''

**Comment by zhangyu2ustc** [comment]
working with batch-size=1, but the training process is too slow.
Need to fix the mini-batch of gcn


Update: fixed the problem. 
there are two ways to doing the batches. the silly way is using a for loop, which is straightforward but very slow.
the other way is the do permutate and view in pytorch

**Comment by zhangyu2ustc** [comment]
https://github.com/meliketoy/graph-cnn.pytorch

**Comment by zhangyu2ustc** [comment]
add LSTM as one of the baseline methods:

In pytorch, the performance of LSTM (5 stacked layers followed by 3 linear layers) are quite good.
for motor classification (epochs=100):
 train_acc: 0.99
 test_acc: 0.95

**Comment by zhangyu2ustc** [comment]
After several experiments on GCN and GAT in pytorch. I find it quite strange that the performance is slightly lower in pytorch implementation.

for motor classification (test_acc):
                          GCN     ChebNet     GAT
tensorflow:        0.98       0.97
pytorch:             0.88       0.92          

we need to figure out what was wrong in the pytorch version

**Comment by zhangyu2ustc** [comment]
transfer learning with different TRs: there are two ways to do it
1) sampling the TR in HCP, which means take the mean of 3 volumes each step
2) interpolation fmri time series in IBC
